libr/search/privkey-find.c

```diff
@@ -71,10 +72,11 @@ R_API int r_search_privkey_update(RSearch *s, ut64 from, const ut8 *buf, int len
 		return -1;
 	}
 	r_list_foreach (s->kws, iter, kw) {
-		for (i = 2; i < len - sizeof (rsa_versionmarker); i++) {
-			if (memcmp (&buf[i], rsa_versionmarker, sizeof (rsa_versionmarker)) && 
-			memcmp (&buf[i], ecc_versionmarker, sizeof (ecc_versionmarker)) &&
-			memcmp (&buf[i], safecurves_versionmarker, sizeof (safecurves_versionmarker))) {
+		// Iteration until the remaining length is too small to contain a key.
+		for (i = 2; i < len - sizeof (rsa_versionmarker) - 4; i++) {
```

Comment:
where is this `4` coming from?

Reply:
I iterate until the remaining length is too small to contain a key.

Reply:
ok but why `4`? Why not 5, 10 or 0? How did you compute this `4`?

Reply:
After the version number you have the integer tag (one byte) and the length which at least one byte but often three, that's where the 4 comes from.

Reply:
Then there should be a better way to write this.
```
 		for (i = 2; i < len - sizeof (rsa_versionmarker) - PRIVKEY_TAG_SIZE - PRIVKEY_LEN_SIZE; i++) {
```
or something like that. Given you are more familiar with the topic you probably can find better names. Also, this is confusing because at https://github.com/radareorg/radare2/pull/17323/files#diff-3c8c44930d4488e02a4930d2003ba188R71 the function checks for `len < sizeof (rsa_versionmarker)` (and not `len < sizeof (rsa_versionmarker) - PRIVKEY_TAG_SIZE - PRIVKEY_LEN_SIZE`, so it's not even said there are enough bytes to make this subtraction work.

Reply:
Now I have defined a constant which is the minimal length for a search and I check the length properly.